packages/blocks/src/api/factory.js

```diff
@@ -77,6 +84,38 @@ export function createBlock( name, attributes = {}, innerBlocks = [] ) {
 	};
 }
 
+/**
+ * Given a map of attributes, returns its subset of transient attributes.
+ *
+ * "Transient attributes", declared in the block type with the `transient:
+ * true` property, represent a special kind of attributes that:
+ *
+ * - are ignored during block serialization;
+ * - don't make their way to the editor's edit history.
+ *
+ * Combined, these properties make transient attributes useful cues for blocks
+ * that need to manage data resolution on their own. Without it, these
+ * resolutions create unwanted undo levels that can easily leave the editor in
+ * a state out of which a user cannot get out.
+ *
+ * For instance, a [gallery] shortcode can be converted to a Gallery block, but
+ * the latter needs to take the shortcode's image IDs, request image data from
+ * the media endpoint, and upon resolution update its own attributes to include
+ * image source and alt text. Thanks to its `transientImageIds` transient
+ * attribute, triggering Undo doesn't cause the block to retrigger this change.
+ *
+ * @param {Object} blockType  Block type.
+ * @param {Object} attributes Block attributes.
+ *
+ * @return {Object} "Transient" block attributes.
+ */
+export function getTransientAttributes( blockType, attributes ) {
```

Comment:
Your branch is probably stale. Most functions accept `blockTypeOrName` as argument.

See: #11490